### DISCUSSION TITLE
Issue with Absolute return and (maybe) XIRR.

### DISCUSSION THREAD

**Discussion by avinashkanaujiya** [ananthakumaran/paisa#45] [Q&A]
In my assets dashboard, after importing my zerodha tradebook, the absolute return shown is really less, it is taking into account the investment amount column, due to which investment is looking really small.

E.g I have 10Lakh in capital, and I traded using that amount for about a year and then at the end, I have investment amount of 2cr and withdrawal amount of 1.85cr, and unbooked gains of 2Lakh.

Now what should be the absolute return? Because I don't think that only investment amount column, should be considered, gains should be looked at as a percentage of capital is present in the moment i.e market value.

**Comment by ananthakumaran** [comment]
I have been going back and forth with how some of the values are calculated (Net Investment, Absolute Returns, Gain Loss). What makes sense in one scenario doesn't make in another. In particular, paisa is written with investors in mind, so trading might not work well.

To make the discussion more concrete, could you put a very small example ledger file and tell me what your expected absolute return, etc. should be?

**Comment by avinashkanaujiya** [comment]
I will try to provide you with some example ledger file.
Also, I don't trade very often, I buy and sell in about 6 to 12 months, so not that often as typical traders would do. And to calculate Absolute Return, I think we should only look at currently existing total capital for that specific account, because according to my experience asset dashboard only shows unbooked gains for assets, and XIRR & Absolute Return is based on those unbooked gains.
Also am i right to suggest that only unbooked gains shows up, or it is the total historical gains, if i am importing the whole tradebook worth of 3 years data?

**Reply by ananthakumaran** [reply]
* XIRR should work regardless of how things are done, so what you see is likely correct, it takes all the transactions into consideration
* Gain is `current value - (invested - withdrawal)`
* Absolute returns `(current value - (invested - withdrawal)) / invested`

Is the contention about how withdrawal is handled? 

In trading, since you would be investing & withdrawing a lot, absolute returns would be small, since the gain is divided by total investment.
